lib/Doctrine/ORM/Query/Lexer.php

```diff
@@ -175,7 +178,13 @@ protected function getType(&$value)
                     }
                 }
 
-                return self::T_IDENTIFIER;
+                if (strpos($value, '\\') !== false) {
+                    return self::T_QUALIFIED_NAME;
+                } else if (strpos($value, ':') !== false) {
```

Comment:
Also, shouldn't this condition come first, since any name containing a `:` is an alias?